__tests__/Unit/Components/Tasks/TaskDetails.test.tsx

```diff
@@ -309,6 +300,36 @@ describe('Textarea with functionalities', () => {
     });
 });
 
+describe('Update Progress button', () => {
+    it('renders the Update Progress button when ?dev=true query parameter is present', async () => {
+        renderWithRouter(
+            <Provider store={store()}>
+                <TaskDetails taskID={details.taskID} />
+            </Provider>,
+            { query: { dev: 'true' }, push: mockNavigateToUpdateProgressPage }
+        );
+
+        await waitFor(() => {
+            const updateProgressButton = screen.getByText('Update Progress');
+            expect(updateProgressButton).toBeInTheDocument();
+            fireEvent.click(updateProgressButton);
+            expect(mockNavigateToUpdateProgressPage).toHaveBeenLastCalledWith(
+                '/progress/6KhcLU3yr45dzjQIVm0J?dev=true'
+            );
+        });
+    });
+
+    it('Should not render the Update Progress button when ?dev=true query parameter is absent', () => {
+        renderWithRouter(
+            <Provider store={store()}>
+                <TaskDetails taskID={details.taskID} />
+            </Provider>
+        );
+        const updateProgressButton = screen.queryByText('Update Progress');
+        expect(updateProgressButton).toBeNull();
```

Comment:
Any specific reason to use ToBeNUll and not using
expect().not.toBeInTheDocument()

Reply:
No specific reason for using it, I have updated it to `expect().not.toBeInTheDocument()` .
May I know the reason for using `expect().not.toBeInTheDocument()` over `expect().toBeNull()`, It would be very helpful if you can provide any resource related to this preference, Thank you!